schema/manager.go

```diff
@@ -136,6 +136,68 @@ func (manager *Manager) OrderedSchemas() []*Schema {
 	return res
 }
 
+//reorderSchema tries reorder schemas using Tarjan's algorithm
+type state int
+
+const (
+	notVisited state = iota
+	visited
+	temporaryVisited
+)
+
+func visitSchema(schemas []*Schema, schemaList []string, schema *Schema, visitedSchema map[string]state) ([]string, error) {
+	if visitedSchema[schema.ID] == temporaryVisited {
+		return nil, fmt.Errorf("Schemas aren't DAG. We can't reorder automatically")
+	}
+	if visitedSchema[schema.ID] == visited {
+		return schemaList, nil
```

Comment:
It will be reached.
Code is here
```
function visit(node n)
    if n has a temporary mark then stop (not a DAG)
    if n is not marked (i.e. has not been visited yet) then
```

We will mark each node in three color In Tarjan algorithm.

white: not visited
gray: temporary visited 
black: visited

Note that gray color will be only painted for node where we are checking.
so it is not a DAG if you meet that color again.
If you meet black color, it is no problem because that means that node is already checked

https://commons.wikimedia.org/wiki/File:Tarjan%27s_Algorithm_Animation.gif